core/runtime/ops.rs

```diff
@@ -1165,4 +1324,115 @@ mod tests {
     )?;
     Ok(())
   }
+
+  #[op2(core, async)]
+  async fn op_async_void() {}
+
+  #[tokio::test]
+  pub async fn test_op_async_void() -> Result<(), Box<dyn std::error::Error>> {
+    run_async_test(10000, "op_async_void", "await op_async_void()").await?;
+    Ok(())
+  }
+
+  #[op2(core, async)]
+  async fn op_async_number(x: u32) -> u32 {
+    x
+  }
+
+  #[op2(core, async)]
+  async fn op_async_add(x: u32, y: u32) -> u32 {
+    x + y
+  }
+
+  #[tokio::test]
+  pub async fn test_op_async_number() -> Result<(), Box<dyn std::error::Error>>
+  {
+    run_async_test(
+      10000,
+      "op_async_number",
+      "assert(await op_async_number(__index__) == __index__)",
+    )
+    .await?;
+    run_async_test(
+      10000,
+      "op_async_add",
+      "assert(await op_async_add(__index__, 100) == __index__ + 100)",
+    )
+    .await?;
+    Ok(())
+  }
+
+  #[op2(core, async)]
+  async fn op_async_sleep() {
+    tokio::time::sleep(Duration::from_millis(500)).await
+  }
+
+  #[op2(core, async)]
+  fn op_async_sleep_impl() -> impl Future<Output = ()> {
+    tokio::time::sleep(Duration::from_millis(500))
+  }
+
+  #[tokio::test]
+  pub async fn test_op_async_sleep() -> Result<(), Box<dyn std::error::Error>> {
+    run_async_test(5, "op_async_sleep", "await op_async_sleep()").await?;
+    run_async_test(5, "op_async_sleep_impl", "await op_async_sleep_impl()")
+      .await?;
+    Ok(())
+  }
+
+  #[op2(core, async)]
```

Comment:
Random question: does the order of attributes matter here? Ie. can we do `op2(async, core)` as well?

Reply:
I considered requiring them to be alphabetically sorted but didn't get to it in this PR. I think now that you've pointed it out, I'll make that a requirement (and print the expected order if it is not correct).